### ISSUE TITLE
[BUG] tensorflow.autolog or keras.autolog submitting empty runs

### ISSUE THREAD

**Issue by TUMAAA** [mlflow/mlflow#5169]
Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md) for additional information about bug reports. For help with debugging your code, please refer to [Stack Overflow](https://stackoverflow.com/questions/tagged/mlflow).

**Please fill in this bug report template to ensure a timely and thorough response.**

### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the MLflow code base?

- [ ] Yes. I can contribute a fix for this bug independently.
- [ ] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [ ] No. I cannot contribute a bug fix at this time.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 20.04
- **MLflow installed from (source or binary)**: from binary (pip install)
- **MLflow version (run ``mlflow --version``)**: 1.18.0 (also tried with 1.22.0)
- **Python version**: Python 3.8.10
- **npm version, if running the dev UI**:
- **Exact command to reproduce**: 

### Describe the problem
Using .autolog (mlflow.keras.autolog() or mlflow.tensorflow.autolog()) on a Keras model is submitting empty runs (no params and no metrics)

 ### Code to reproduce issue
```
epoch = self.load_checkpoint(network=self.training_model, directory_helper=self.directory_helper)
self._save_model_description()
checkpoint_path = os.path.join(self.directory_helper.checkpoint_dir(), CHECKPOINT_PATH_PATTERN)
save_callback = ModelCheckpoint(
            filepath=checkpoint_path, save_weights_only=True, save_freq="epoch", monitor="loss"
        )
MLFLOW_TRACKING_SERVER_INSTANCE = "http://localhost:5000"
mlflow.set_tracking_uri(MLFLOW_TRACKING_SERVER_INSTANCE)
mlflow.set_experiment("my-test")
mlflow.tensorflow.autolog()
# mlflow.keras.autolog() # tried also this
with mlflow.start_run() as run:
    print("Active run_id: {}".format(run.info.run_id)) # giving me a valid id
    self.training_model.fit(
        sample_iterator.framework_iterable(),
        epochs=self.epochs,
        steps_per_epoch=self.steps_per_epoch,
        callbacks=self.callbacks + [save_callback],
        initial_epoch=epoch,
    )
    self.training_model.evaluate(sample_iterator.framework_iterable(), steps=1)
    self.training_model.save("/tmp/model")
```
The code runs successfully (train, evaluate and a model is saved)

### Other info / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

Testing started at 14:48 ...
/<somedir>/bin/python /home/<somedir>/pycharm-2020.2.2/plugins/python/helpers/pycharm/_jb_pytest_runner.py --target network_integration_test.py::test_training_integration
Launching pytest with arguments network_integration_test.py::test_training_integration in <script>

============================= test session starts ==============================
platform linux -- Python 3.8.10, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- /<somedir>/bin/python
cachedir: .pytest_cache
rootdir: <script_dir>
collecting ... collected 1 item

network_integration_test.py::test_training_integration PASSED            [100%]Using data set train and preprocessor mode train_steps.
Active run_id: 27da2a3697764209b64851b9b3f95c1f
1/1 [==============================] - 0s 19ms/step - batch: 0.0000e+00 - size: 1.0000 - loss: 4113.8296
Training is done.
waiting 2s---------------------------
Using data set validation and preprocessor mode validation_steps.
inferred output:
 {'image_extractor': {'output': array([[[0.0000000e+00, 6.8138964e-03, 0.0000000e+00],
        [0.0000000e+00, 1.1089750e-01, 0.0000000e+00],
        [0.0000000e+00, 1.4115120e-01, 0.0000000e+00],
        ...,
        [0.0000000e+00, 4.0602821e+01, 0.0000000e+00],
        [0.0000000e+00, 7.1283195e+01, 0.0000000e+00],
        [0.0000000e+00, 7.4124146e+01, 0.0000000e+00]],

       [[0.0000000e+00, 2.6389563e-01, 0.0000000e+00],
        [0.0000000e+00, 6.3583678e-01, 0.0000000e+00],
        [0.0000000e+00, 9.5497471e-01, 0.0000000e+00],
        ...,
        [0.0000000e+00, 8.9535469e+01, 0.0000000e+00],
        [0.0000000e+00, 8.9088066e+01, 0.0000000e+00],
        [0.0000000e+00, 8.9849754e+01, 0.0000000e+00]],

       [[0.0000000e+00, 2.4204983e-01, 0.0000000e+00],
        [0.0000000e+00, 4.7542697e-01, 0.0000000e+00],
        [0.0000000e+00, 7.0950073e-01, 0.0000000e+00],
        ...,
        [0.0000000e+00, 5.1282654e+01, 0.0000000e+00],
        [7.3204637e+00, 4.3331047e+01, 0.0000000e+00],
        [0.0000000e+00, 5.6928738e+01, 0.0000000e+00]],

       ...,

       [[0.0000000e+00, 1.8032816e-01, 0.0000000e+00],
        [0.0000000e+00, 3.6331969e-01, 0.0000000e+00],
        [0.0000000e+00, 5.4345101e-01, 0.0000000e+00],
        ...,
        [0.0000000e+00, 1.9824995e+01, 0.0000000e+00],
        [0.0000000e+00, 1.1003504e+01, 0.0000000e+00],
        [0.0000000e+00, 3.6498924e+01, 0.0000000e+00]],

       [[0.0000000e+00, 7.8257382e-02, 0.0000000e+00],
        [0.0000000e+00, 1.9469540e-01, 0.0000000e+00],
        [0.0000000e+00, 2.3878448e-01, 0.0000000e+00],
        ...,
        [0.0000000e+00, 2.6312938e+00, 0.0000000e+00],
        [9.3650808e+00, 0.0000000e+00, 0.0000000e+00],
        [1.0873189e+01, 8.4439039e+00, 0.0000000e+00]],

       [[0.0000000e+00, 2.2829400e-02, 0.0000000e+00],
        [0.0000000e+00, 8.9347780e-02, 0.0000000e+00],
        [0.0000000e+00, 1.3833980e-01, 0.0000000e+00],
        ...,
        [5.5682340e+00, 6.9527259e+00, 0.0000000e+00],
        [1.1677513e+01, 0.0000000e+00, 0.0000000e+00],
        [5.9880052e+00, 0.0000000e+00, 0.0000000e+00]]], dtype=float32)}}
2021-12-15 14:48:59.091233: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcuda.so.1
2021-12-15 14:48:59.112495: E tensorflow/stream_executor/cuda/cuda_driver.cc:314] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
2021-12-15 14:48:59.112559: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: cmucl855212
2021-12-15 14:48:59.112570: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: cmucl855212
2021-12-15 14:48:59.112671: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: 470.57.2
2021-12-15 14:48:59.112713: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 470.57.2
2021-12-15 14:48:59.112720: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:310] kernel version seems to match DSO: 470.57.2
2021-12-15 14:48:59.112968: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN)to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-12-15 14:48:59.136761: I tensorflow/core/platform/profile_utils/cpu_utils.cc:104] CPU Frequency: 2599990000 Hz
2021-12-15 14:48:59.138434: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x2302b70 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2021-12-15 14:48:59.138455: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2021-12-15 14:48:59.961617: W tensorflow/python/util/util.cc:348] Sets are not currently considered sequences, but this may change in the future, so consider avoiding using them.

=============================== warnings summary ===============================
../../../../../../../../../venv/lib/python3.8/site-packages/tensorflow/python/autograph/utils/testing.py:21
  /<somedir>/lib/python3.8/site-packages/tensorflow/python/autograph/utils/testing.py:21: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

network_integration_test.py::test_training_integration
  /<somedir>/lib/python3.8/site-packages/pkg_resources/__init__.py:1144: DeprecationWarning: Use of .. or absolute path in a resource path is not allowed and will raise exceptions in a future release.
    return get_provider(package_or_requirement).get_resource_filename(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
======================== 1 passed, 2 warnings in 5.24s =========================

Process finished with exit code 0

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [x] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [x] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by TUMAAA** [comment]
Any answer? :)

**Comment by TUMAAA** [comment]
After a lot of investigation here is the answer:
**When tensorflow eager mode is disabled the autolog submits empty results.** 
To verify simply take the (sample code)[examples/keras/train.py] mlflow github and disable eager mode by simply adding tensorflow.compat.v1.disable_eager_execution(). You will notice the results submitted are empty

**Comment by dbczumar** [comment]
@TUMAAA Apologies for the delayed response. Thank you for investigating. We are working on a fix.
